File: Week 5/Python Codes/RemoveDuplicatesFromSortedArrayII.py
class Solution:
    # @param A : list of integers
    # @return an integer
    def removeDuplicates(self, A):
        count, size = 0, len(A)
        for i in range(0, size):
            if (i<size-2 and A[i]==A[i+1] and A[i]==A[i+2]):
                continue
            else:
                A[count] = A[i]
                count += 1
        return count

File: Week 5/C++ Codes/RemoveDuplicatesFromSortedArrayII.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    public:
        int removeDuplicates(vector<int> &A) {
        // Do not write main() function.
        // Do not read input, instead use the arguments to the function.
        // Do not print the output, instead return values as specified
        // Still have a doubt. Checkout www.interviewbit.com/pages/sample_codes/ for more details
        int count = 0, size = A.size();
        for(auto i=0; i<size; i++) {
            if (i<size-2 && A[i]==A[i+1] && A[i]==A[i+2]) continue;
            else
                A[count++] = A[i];
        }
        return count;
        }
};